osu.Game/Overlays/Profile/Sections/Ranks/DrawableScore.cs

```diff
@@ -0,0 +1,213 @@
+            {
+                AutoSizeAxes = Axes.Both,
+                Url = $"https://osu.ppy.sh/beatmaps/{score.Beatmap.OnlineBeatmapID}",
+                Child = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new OsuSpriteText
+                        {
+                            Current = locale.GetUnicodePreference($"{score.Beatmap.Metadata.TitleUnicode ?? score.Beatmap.Metadata.Title} [{score.Beatmap.Version}] ", $"{score.Beatmap.Metadata.Title ?? score.Beatmap.Metadata.TitleUnicode} [{score.Beatmap.Version}] "),
+                            TextSize = 15,
+                            Font = "Exo2.0-SemiBoldItalic",
+                        },
+                        new OsuSpriteText
+                        {
+                            Current = locale.GetUnicodePreference(score.Beatmap.Metadata.ArtistUnicode, score.Beatmap.Metadata.Artist),
+                            TextSize = 12,
+                            Padding = new MarginPadding { Top = 3 },
+                            Font = "Exo2.0-RegularItalic",
+                        },
+                    },
+                },
+            });
+
+            foreach (Mod mod in score.Mods)
+                modContainer.Add(new ModIcon(mod.Icon, colour.Yellow));
+        }
+
+        private class ModContainer : FlowContainer<ModIcon>
+        {
+            protected override IEnumerable<Vector2> ComputeLayoutPositions()
+            {
+                int count = FlowingChildren.Count();
+                for (int i = 0; i < count; i++)
+                    yield return new Vector2(DrawWidth * i * (count == 1 ? 0 : 1f / (count - 1)), 0);
+            }
+        }
+
+        private class ModIcon : Container
+        {
+            public ModIcon(FontAwesome icon, Color4 colour)
+            {
+                AutoSizeAxes = Axes.Both;
+
+                Children = new[]
+                {
+                    new TextAwesome
+                    {
+                        Origin = Anchor.Centre,
+                        Anchor = Anchor.Centre,
+                        Icon = FontAwesome.fa_osu_mod_bg,
+                        Colour = colour,
+                        Shadow = true,
+                        TextSize = 30,
+                        UseFullGlyphHeight = false,
+                    },
+                    new TextAwesome
+                    {
+                        Origin = Anchor.Centre,
+                        Anchor = Anchor.Centre,
+                        Icon = icon,
+                        Colour = OsuColour.Gray(84),
+                        TextSize = 18,
+                        Position = new Vector2(0f, 2f),
+                        UseFullGlyphHeight = false,
+                    },
+                };
+            }
+        }
+
+        private class LinkContainer : OsuClickableContainer

```

Comment:
Why couldn't you use the existing LinkText implementation? Code duplication is bad.